docs/docs/tutorials/orca-yarn-tutorial.md

```diff
@@ -0,0 +1,601 @@
+
+# 2. Prepare Environment
+Before running the BigDL program on Yarn, we need to setup the environment as the following steps:
+* Setup JAVA & Hadoop Environment
+* Install All Needed Python Libraries 
+* For Spark Installed Cluster(CDH Users)
+
+## 2.1 Setup JAVA & Hadoop Environment
+We need to download and install JDK in the environment, and properly set the environment variable JAVA_HOME, which is required by Spark. JDK8 is highly recommended.
+
+### 2.1.1 Setup JAVA Environment
+```bash
+# For Ubuntu
+sudo apt-get install openjdk-8-jre
+export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64/
+
+# For CentOS
+su -c "yum install java-1.8.0-openjdk"
+export JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.282.b08-1.el7_9.x86_64/jre
+
+export PATH=$PATH:$JAVA_HOME/bin
+java -version  # Verify the version of JDK.
+```
+
+### 2.1.2 Setup Hadoop Environment
+Check the Hadoop setup and configurations of our cluster. Make sure we correctly set the environment variable `HADOOP_CONF_DIR`, which is needed to initialize Spark on YARN:
+```bash
+export HADOOP_CONF_DIR=/path/to/hadoop/conf
+```
+
+## 2.2 Install Needed Python Libraries
+### 2.2.1 Install and Create Conda Environment
+We need first to use conda to prepare the Python environment on the local machine where we submit our application. You could download and install Conda following [Conda User Guide](https://docs.conda.io/projects/conda/en/latest/user-guide/install/linux.html) or executing the command as below.
+```bash
+# Download Anaconda installation script 
+wget -P /tmp https://repo.anaconda.com/archive/Anaconda3-2020.02-Linux-x86_64.sh
+
+# Execute the script to install conda
+bash /tmp/Anaconda3-2020.02-Linux-x86_64.sh
+
+# Please type this command in your terminal to activate Conda environment
+source ~/.bashrc
+``` 
+
+### 2.2.2 Use Conda to Install BigDL
+Create a conda environment, install BigDL and all the needed Python libraries in the created conda environment:
+``` bash
+# "bigdl" is conda environment name, you can use any name you like.
+# Please change Python version to 3.6 if you need a Python 3.6 environment.
+conda create -n bigdl python=3.7 
+conda activate bigdl
+```
+You can install the latest release version of BigDL (built on top of Spark 2.4.6 by default) as follows:
+```bash
+pip install bigdl
+```
+You can install the latest nightly build of BigDL as follows:
+```bash
+pip install --pre --upgrade bigdl
+```
+Using Conda to install BigDL will automatically install libraries including:
+```bash
+conda-pack              0.3.1
+numpy                   1.21.6
+pandas                  1.2.5
+pyarrow                 7.0.0
+pyspark                 2.4.6
+pytorch-lightning       1.4.2
+tensorboard             2.9.1
+torch                   1.9.0
+torchmetrics            0.9.1
+torchvision             0.10.0
+redis                   4.3.3
+scikit-learn            1.0.2
+...
```

Comment:
Instead of listing all the details, just highlight some important dependency, as the details may change in future versions.